src/test/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/LaboratoryInvestigationResultTest.java

```diff
@@ -0,0 +1,105 @@
+    }
+
+    @Test
+    void when_edifactString2IsPassed_expect_returnALaboratoryInvestigationResultObject() {
+        assertThat(laboratoryInvestigationResult2)
+                .usingRecursiveComparison()
+                .isEqualTo(LaboratoryInvestigationResult.fromString("RSL+NV+11.9++:::ng/mL+HI"));
+    }
+
+    @Test
+    void when_edifactString3IsPassed_expect_returnALaboratoryInvestigationResultObject() {
+        assertThat(laboratoryInvestigationResult3)
+                .usingRecursiveComparison()
+                .isEqualTo(LaboratoryInvestigationResult.fromString("RSL+NV+11.9++:::ng/mL"));
+    }
+
+    @Test
+    void when_mappingSegmentObjectToEdifactString_expect_returnCorrectEdifactString() {
+        String expectedEdifactString = "RSL+NV+11.9:7++:::ng/mL+HI'";
+
+        LaboratoryInvestigationResult laboratoryInvestigationResult = LaboratoryInvestigationResult.builder()
+            .measurementValue(new BigDecimal("11.9"))
+            .measurementValueComparator(MeasurementValueComparator.LESS_THAN)
+            .measurementUnit("ng/mL")
+            .deviatingResultIndicator(DeviatingResultIndicator.ABOVE_HIGH_REFERENCE_LIMIT)
+            .build();
+
+        assertEquals(expectedEdifactString, laboratoryInvestigationResult.toEdifact());
+    }
+
+    @Test
+    void when_buildingSegmentObjectWithoutAnyFields_expect_nullPointerExceptionIsThrown() {
+        assertThrows(NullPointerException.class, () -> LaboratoryInvestigation.builder().build());
+    }
+
+    @Test
+    void testGetKey() {
+        assertEquals(laboratoryInvestigationResult1.getKey(), "RSL");
+    }
+
+    @Test
+    void testGetValue() {
+        assertEquals(laboratoryInvestigationResult1.getValue(), "NV+11.9:7++:::ng/mL+HI");
+    }
+
+    @Test
+    void testValidateStateful() {
+        assertDoesNotThrow(laboratoryInvestigationResult1::validateStateful);
+    }
+
+    @Test
+    void testPreValidate() {
+        LaboratoryInvestigationResult emptyMeasurementValue = new LaboratoryInvestigationResult(
+                null, MeasurementValueComparator.LESS_THAN, "ng/mL", DeviatingResultIndicator.ABOVE_HIGH_REFERENCE_LIMIT
+        );
+        LaboratoryInvestigationResult emptyMeasurementUnit = new LaboratoryInvestigationResult(
+                new BigDecimal("11.9"), MeasurementValueComparator.LESS_THAN, null, DeviatingResultIndicator.ABOVE_HIGH_REFERENCE_LIMIT
+        );
+
+        assertSoftly(softly -> {
+            softly.assertThatThrownBy(emptyMeasurementValue::preValidate)
+                    .isExactlyInstanceOf(EdifactValidationException.class)
+                    .hasMessage("RSL: Attribute measurementValue is required");
+
+            softly.assertThatThrownBy(emptyMeasurementUnit::preValidate)
+                    .isExactlyInstanceOf(EdifactValidationException.class)
+                    .hasMessage("RSL: Attribute measurementUnit is required");
+        });
```

Comment:
I find the softly stuff from assertj distracting, especially when junit5 gives us `assertAll` anyway.

```suggestion
        assertAll(
            () -> assertThatThrownBy(emptyMeasurementValue::preValidate)
                    .isExactlyInstanceOf(EdifactValidationException.class)
                    .hasMessage("RSL: Attribute measurementValue is required"),
            () -> assertThatThrownBy(emptyMeasurementUnit::preValidate)
                    .isExactlyInstanceOf(EdifactValidationException.class)
                    .hasMessage("RSL: Attribute measurementUnit is required")
        );
```